packages/opal-common/opal_common/git/commit_viewer.py

```diff
@@ -177,16 +175,12 @@ def files(
         Yields:
             the next file found (only for files passing the filter).
         """
-        files_generator = self.nodes(lambda node: isinstance(node, VersionedFile))
-        if filter is None:
-            yield from files_generator
-        else:
-            for f in files_generator:
-                if filter(f):
-                    yield f
+        return filter(
+            lambda node: isinstance(node, VersionedFile), self.nodes(predicate)
+        )
```

Comment:
Nitpick: this could be written more clearly as a generator expression:
```suggestion
        return (
            node for node in self.nodes(predicate)
            if isinstance(node, VersionedFile)
        )
```